python/nav/web/netmap/graph.py

```diff
@@ -176,17 +218,32 @@ def get_layer2_traffic():
         })
 
     _logger.debug('Time used: %s', datetime.now() - start)
+    cache.set(cache_key, traffic, TRAFFIC_CACHE_TIMEOUT)
     return traffic
 
-
-def get_layer3_traffic():
+def get_layer3_traffic(locationOrRoomId):
     """Fetches traffic data for layer 3"""
+
+    # Cache model: Index traffic by location
+    cache_key = _cache_key("traffic", "locationId", "layer 3")
```

Comment:
Yeah, turns out I forgot to actually cache on the location properly. I don't really know what happened here. Should be fixed now.